### DISCUSSION TITLE
How to filter types (platform/library/user) using API?

### DISCUSSION THREAD

**Discussion by saruman9** [Vector35/binaryninja-api#3019] [API] [answered]
For example, I want to create a type library for a binary with DWARF, but I don't want to include system types (platform types). How can I get only user defined types using API?

**Comment by fuzyll** [comment]
I think this might've been answered by this: https://github.com/Vector35/binaryninja-api/discussions/3020#discussioncomment-2381000.

Is that sufficient? Or do you need more information?

**Reply by saruman9** [reply]
Hm, I didn't find the answer. `Type` class didn't have `auto` property.

I will try to explain what I need. For example, we have three types (`current_view.types`): default `Elf32_Dyn`, user-defined `my_structure` and `std::__cxx11::string` from type library (`libstdc++_i386.so.6`). How can I get information that `my_structure` is user-defined, `Elf32_Dyn` is auto type and `std::__cxx11::string` from concrete type library?

**Comment by saruman9** [comment, marked as answer]
I found needed method: [`is_type_auto_defined`](https://api.binary.ninja/binaryninja.binaryview-module.html#binaryninja.binaryview.BinaryView.is_type_auto_defined).
